### ISSUE TITLE
Join forces with HumanReadableIDs?

### ISSUE THREAD

**Issue by c42f** [pbayer/Proquint.jl#3]
Hi, nice package!

I didn't see this package so I coincidentally implemented the same thing over at HumanReadableIDs.jl (an unregistered package):

https://github.com/c42f/HumanReadableIDs.jl/blame/main/src/proquints.jl

My implementation seems to be between 3x and 10x faster though — would you consider just importing the code from there?

```julia
julia> @btime uint2quint(0x7f000001)
  150.438 ns (7 allocations: 384 bytes)
"lusab-babad"

julia> @btime HumanReadableIDs.encode_proquint(0x7f000001)
  46.484 ns (2 allocations: 128 bytes)
"lusab-babad"

julia> x = rand(UInt128)
0x4e271a8420f8196877232885ffaf029c

julia> @btime uint2quint($x)
  860.741 ns (31 allocations: 1.84 KiB)
"humol-dopah-fagum-dojom-lisog-fofaj-zuvoz-bapis"

julia> @btime HumanReadableIDs.encode_proquint($x)
  87.224 ns (2 allocations: 192 bytes)
"humol-dopah-fagum-dojom-lisog-fofaj-zuvoz-bapis"
```

Also I think it would be useful if UUIDs were supported via the UInt128 code path.

**Comment by pbayer** [comment]
Hi, 
thank you! I'd love to join forces and import your code.

Currently I use the package in Actors.jl to create a human readable identifier for an actor (a task address). Since in encoding addresses my "short form" is convenient, I would have to modify your code for that purpose if you don't mind.

> Also I think it would be useful if UUIDs were supported via the UInt128 code path.

Yes, that makes sense. 

You can do also a PR if you like.

**Comment by c42f** [comment]
> You can do also a PR if you like.

Sure, I'll look into that. I just wanted to check you were ok for me to replace most of the code here!

(BTW I keep typo'ing the package name as `Proquints.jl` as plural package names are a pretty strong Julia convention. Not sure it's easy to do anything about that though.)

**Comment by pbayer** [comment]
> I just wanted to check you were ok for me to replace most of the code 

I would welcome it. I did not invest much in that code and I appreciate very much a better version of it. 

You are right that `Proquints.jl` would be better. Do you think, it would be worth to rename the package?

**Comment by c42f** [comment]
> Do you think, it would be worth to rename the package?

Good question. I had a look for whether renaming is possible,  but all I found was
* https://github.com/JuliaRegistries/General/issues/2770
* https://discourse.julialang.org/t/procedures-for-package-renaming-deletion-and-deprecation/48108/11

which suggests that creating a separate package with the new name and new package UUID is the way to do it. So it's kind of possible but not exactly easy or neat! Perhaps not worth the effort.

**Comment by pbayer** [comment]
By the same assumptions we are holding when we provide and develop that package, namely that
1. it is useful (and will become more so over time as numbers in computing get bigger) and
2. Julia has a future in computing 
 
and with `Proquints` being more idiomatic to Julia it makes sense to rename the package to `Proquints.jl`. 

At the moment only my own packages at [JuliaActors](https://github.com/JuliaActors) depend from it. So I can rename it without creating too much disturbance. I will do it in the next few weeks. 

Please wait with your PR until it has been renamed. (I open a new issue for renaming).

**Comment by c42f** [comment]
> Please wait with your PR until it has been renamed

If you rename the repo within the github UI, any old issues/PRs will be preserved and any old links to `Proquint.jl` will redirect to `Proquints.jl`.

**Comment by pbayer** [comment]
Yes, that is true. We won't loose anything. If you have yet prepared your PR just do it.

Renaming a package is easier with version 0.1.0.  Also I have to edit the source code. 

For clarity your code will go into 0.2.0. So I think it is easier if we do it in two steps.

But since this is a simple package it won't matter much.

**Comment by c42f** [comment]
Sure don't hold anything up on my account. I was just pointing out that there's a good workflow for renaming on the github side (if not in the registry)
